Doc/library/importlib.resources.rst

```diff
@@ -33,47 +35,37 @@ not** have to exist as physical files and directories on the file system.
    on `using importlib.resources
    <http://importlib-resources.readthedocs.io/en/latest/using.html>`_ and
    `migrating from pkg_resources to importlib.resources
-   <http://importlib-resources.readthedocs.io/en/latest/migration.html>`_
-   and
-   `migrating legacy usage <https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy>`_.
+   <http://importlib-resources.readthedocs.io/en/latest/migration.html>`_.
 
-Loaders that wish to support resource reading should implement a
+:class:`Loaders <importlib.abc.Loader>` that wish to support resource reading should implement a
 ``get_resource_reader(fullname)`` method as specified by
-:class:`importlib.abc.ResourceReader`.
-
-The following types are defined.
+:class:`importlib.resources.abc.ResourceReader`.
 
 .. data:: Package
 
-    The ``Package`` type is defined as ``Union[str, ModuleType]``.  This means
-    that where the function describes accepting a ``Package``, you can pass in
-    either a string or a module.  Module objects must have a resolvable
-    ``__spec__.submodule_search_locations`` that is not ``None``.
-
-.. data:: Resource
-
-    This type describes the resource names passed into the various functions
-    in this package.  This is defined as ``Union[str, os.PathLike]``.
+    Where a function is described as accepting a ``Package``, you can pass in
+    either a module name (as a string) or a module object. Module objects must
+    have a resolvable ``__spec__.submodule_search_locations`` that is
+    not ``None``.
 
-
-The following functions are available.
+    The ``Package`` type is defined as ``Union[str, ModuleType]``.
```

Comment:
PEP-604 notation:
```suggestion
    The ``Package`` type is defined as ``str | ModuleType``.
```

Reply:
That leaves users unfamiliar with typing without a word to search for. I vote against this change.